src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -100,6 +104,14 @@ private void UpdateParameterMetadata(ModelMetadata metadata, IBinderMetadata bin
                     modelBindingContext.IsModelSet)
                 {
                     arguments[parameter.PropertyName] = modelBindingContext.Model;
+                    var validationContext = new ModelValidationContext(
+                    _modelMetadataProvider,
```

Comment:
Looked at it, while we can get away by adding it as an explicit service, I think it is better to leave it on the context as it comes in handy if the validator needs to lookup a metadata which is not related to the model directly ( like a generic type parameter of a property). Leaving this as is. Let me know if you think otherwise.